appchain-anchor/src/user_actions/staking.rs

```diff
@@ -83,8 +83,23 @@ impl AppchainAnchor {
         let protocol_settings = self.protocol_settings.get().unwrap();
         assert!(
             next_validator_set.validator_count() < protocol_settings.maximum_validator_count.0,
-            "Too many validators registered."
+            "The count of registered validators exceeds the upper limit."
         );
+        //
+        if self.appchain_state.eq(&AppchainState::Active) {
+            let validator_set_histories = self.validator_set_histories.get().unwrap();
+            let latest_validator_set = validator_set_histories
+                .get(&validator_set_histories.end_index)
+                .unwrap();
+            if latest_validator_set.validator_count() >= 12 {
+                assert!(
+                    next_validator_set.validator_count()
+                        <= latest_validator_set.validator_count() * 5 / 4,
```

Comment:
The `latest_validator_set.validator_count()` return a u64, then u64 * 5 / 4 is staill a u64,eg: 5 as u64 *5/4 = 6.
So maybe it should change to:
```
next_validator_set.validator_count() * 4
                        <= latest_validator_set.validator_count() * 5 ,
```